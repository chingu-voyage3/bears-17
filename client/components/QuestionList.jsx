/* eslint-disable no-underscore-dangle */
import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

import FormatDate from 'Components/FormatDate';
import UserBar from 'Components/UserBar';

const QuestionList = ({ questions }) => {
  const list = questions.map(question => (
    <li className="question" key={question._id}>
      <h3 className="question__title">
        <Link to={`/question/${question._id}`}>{question.title}</Link>
      </h3>

      <footer className="question__meta">
        <UserBar name={question.author.name} avatar={question.author.avatar} />
        <div className="question__votes">{question.votes} votes</div>
        <span className="question__time">
          <FormatDate date={question.submitted_at} />
        </span>
        <span>X answers</span>
      </footer>
    </li>
  ));

  return (
    <section className="">
      <div className="wrapper">
        <ul className="question-list">
          {list}
        </ul>
      </div>
    </section>
  );
};

QuestionList.defaultProps = {
  questions: [],
};

QuestionList.propTypes = {
  questions: PropTypes.arrayOf(PropTypes.object.isRequired),
};

export default QuestionList;
