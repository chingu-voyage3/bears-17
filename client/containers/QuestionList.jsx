/* eslint-disable no-console */
/* eslint-disable no-underscore-dangle */
/* eslint-disable max-len */
import React, { Component } from 'react';
import List from '../components/QuestionList.jsx';

const pageNav = {
  display: 'flex',
  flexDirection: 'row',
  justifyContent: 'center',
  alignItems: 'center',
};

class QuestionList extends Component {
  constructor(props) {
    super(props);

    this.state = {
      questions: [],
      totalQuestions: 0,
    };
  }
  componentWillMount() {
    this.getQuestionCount();
    this.getQuestions();
  }
  getQuestionCount() {
    fetch('/api/questions/total')
      .then(res => res.json())
      .then((data) => {
        this.setState({ totalQuestions: data });
      });
  }
  getQuestions(page) {
    const url = page ? `/api/questions?page=${page}` : '/api/questions';
    fetch(url)
      .then(res => res.json())
      .then(data => this.setState({ questions: data }));
  }

  render() {
    const pages = [];
    for (let i = 0; i < this.state.totalQuestions / 10; i += 1) {
      pages.push(<button className="btn btn--outline btn--primary" key={i} onClick={() => this.getQuestions(i + 1)}>{i + 1}</button>);
    }
    return (
      <div>
        <List questions={this.state.questions} />
        <div style={pageNav}>
          {pages}
        </div>
      </div>
    );
  }
}

export default QuestionList;
