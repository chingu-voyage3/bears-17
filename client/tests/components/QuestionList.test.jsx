import React from 'react';
import renderer from 'react-test-renderer';

import QuestionList from 'Components/QuestionList';

const questions = [{
  _id: '5a3304d3fc13ae261a0002c6',
  submitted_at: '2017-12-08T08:23:31Z',
  title: 'Vestibulum ac est lacinia nisi venenatis tristique?',
  body:
    'Curabitur convallis. Duis consequat dui nec nisi volutpat eleifend. Donec ut dolor. Morbi vel lectus in quam fringilla rhoncus.',
  votes: 34,
  author: {
    _id: '5a3304d3fc13ae261a0002c7',
    name: 'Brandais Frodsam',
    avatar:
      'http://dummyimage.com/512x512.jpg/dddddd/000000&text=Brandais Frodsam',
  },
}, {
  _id: '5a3304d3fc13ae261a0002c8',
  submitted_at: '2017-12-04T22:09:29Z',
  title: 'Nulla mollis molestie lorem?',
  body:
    'Aenean sit amet justo. Morbi ut odio. Cras mi pede, malesuada in, imperdiet et, commodo vulputate, justo.',
  votes: 5,
  author: {
    _id: '5a3304d3fc13ae261a0002c9',
    name: 'Matthias Dymock',
    avatar:
      'http://dummyimage.com/512x512.jpg/cc0000/ffffff&text=Matthias Dymock',
  },
}, {
  _id: '5a3304d3fc13ae261a0002ca',
  submitted_at: '2017-12-06T22:20:15Z',
  title: 'Nullam sit amet turpis elementum ligula vehicula consequat?',
  body:
    'Duis consequat dui nec nisi volutpat eleifend. Donec ut dolor. Morbi vel lectus in quam fringilla rhoncus. Mauris enim leo, rhoncus sed, vestibulum sit amet, cursus id, turpis.',
  votes: 67,
  author: {
    _id: '5a3304d3fc13ae261a0002cb',
    name: 'Ingmar Storey',
    avatar:
      'http://dummyimage.com/512x512.jpg/5fa2dd/ffffff&text=Ingmar Storey',
  },
}];

describe('QuestionList', () => {
  it('should match the snapshot', () => {
    const list = renderer.create(<QuestionList questions={questions} />);

    expect(list.toJSON()).toMatchSnapshot();
  });
});
