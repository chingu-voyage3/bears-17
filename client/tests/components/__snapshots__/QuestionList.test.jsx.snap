// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestionList should match the snapshot 1`] = `
<section
  className=""
>
  <div
    className="wrapper"
  >
    <ul
      className="question-list"
    >
      <li
        className="question"
      >
        <h3
          className="question__title"
        >
          <a
            href="/question/5a3304d3fc13ae261a0002c6"
          >
            Vestibulum ac est lacinia nisi venenatis tristique?
          </a>
        </h3>
        <footer
          className="question__meta"
        >
          <div
            className="user"
          >
            <img
              alt=""
              className="user__avatar"
              src="http://dummyimage.com/512x512.jpg/dddddd/000000&text=Brandais Frodsam"
              width="32px"
            />
            <span
              className="user__name"
            >
              Brandais Frodsam
            </span>
          </div>
          <div
            className="question__votes"
          >
            34
             votes
          </div>
          <span
            className="question__time"
          >
            <time
              dateTime="2017-12-08T08:23:31Z"
            >
              Dec 8, 8:23 AM
            </time>
          </span>
          <span>
            X answers
          </span>
        </footer>
      </li>
      <li
        className="question"
      >
        <h3
          className="question__title"
        >
          <a
            href="/question/5a3304d3fc13ae261a0002c8"
          >
            Nulla mollis molestie lorem?
          </a>
        </h3>
        <footer
          className="question__meta"
        >
          <div
            className="user"
          >
            <img
              alt=""
              className="user__avatar"
              src="http://dummyimage.com/512x512.jpg/cc0000/ffffff&text=Matthias Dymock"
              width="32px"
            />
            <span
              className="user__name"
            >
              Matthias Dymock
            </span>
          </div>
          <div
            className="question__votes"
          >
            5
             votes
          </div>
          <span
            className="question__time"
          >
            <time
              dateTime="2017-12-04T22:09:29Z"
            >
              Dec 4, 10:09 PM
            </time>
          </span>
          <span>
            X answers
          </span>
        </footer>
      </li>
      <li
        className="question"
      >
        <h3
          className="question__title"
        >
          <a
            href="/question/5a3304d3fc13ae261a0002ca"
          >
            Nullam sit amet turpis elementum ligula vehicula consequat?
          </a>
        </h3>
        <footer
          className="question__meta"
        >
          <div
            className="user"
          >
            <img
              alt=""
              className="user__avatar"
              src="http://dummyimage.com/512x512.jpg/5fa2dd/ffffff&text=Ingmar Storey"
              width="32px"
            />
            <span
              className="user__name"
            >
              Ingmar Storey
            </span>
          </div>
          <div
            className="question__votes"
          >
            67
             votes
          </div>
          <span
            className="question__time"
          >
            <time
              dateTime="2017-12-06T22:20:15Z"
            >
              Dec 6, 10:20 PM
            </time>
          </span>
          <span>
            X answers
          </span>
        </footer>
      </li>
    </ul>
  </div>
</section>
`;
